mycore-webtools/src/main/java/org/mycore/webtools/vue/MCRVueRootServlet.java

```diff
@@ -127,6 +146,54 @@ public class MCRVueRootServlet extends MCRContentServlet {
     @Serial
     private static final long serialVersionUID = 1L;
 
+    protected static final String VUE_PASSTHROUGH_PROPERTY_NAME = "MCR.Vue.Properties";
+
+    protected static final String HEADING_INIT_PARAM = "heading";
+    protected static final String PROPERTIES_INIT_PARAM = "properties";
+
+    protected String headingI18nKey;
+    protected Set<String> servletInitPropertiesKeys;
+    private Map<String, String> servletInitProperties;
+
+    private static final Set<Integer> PASSTHROUGH_HTTP_STATUS_CODE_LIST = Set.of(
+        HttpServletResponse.SC_NOT_FOUND,
+        HttpServletResponse.SC_UNAUTHORIZED,
+        HttpServletResponse.SC_FORBIDDEN
+    );
+
+    @Override
+    public void init(ServletConfig config) throws ServletException {
+        super.init(config);
+        headingI18nKey = config.getInitParameter(HEADING_INIT_PARAM);
+
+        String propertiesInitParam = config.getInitParameter(PROPERTIES_INIT_PARAM);
+        servletInitPropertiesKeys = propertiesInitParam != null
+            ? new HashSet<>(List.of(propertiesInitParam.split(",")))
+            : new HashSet<>();
+
+        updateInitProperties();
+
+        MCRConfiguration2.addPropertyChangeEventLister(servletInitPropertiesKeys::contains, this::changeProperty);
+    }
+
+    /**
+     * Resolves the property value from the configuration. If the property is not found, an exception is thrown.
+     * If the property is found, but the value is empty, then the value is still returned.
+     * {@link MCRConfiguration2#getString(String)} would return an empty optional in this case.
+     * @param key the property key
+     * @return the property value, even if it is an empty string
+     */
+    private static String resolveProperty(String key) {
+        return MCRConfiguration2.getOrThrow(key, o -> o);
```

Comment:
An empty value leads to an exception, possibly `MCRConfiguration2#getPropertiesMap` should be used.